### PR TITLE
feat: allowing walStorage to be changed on existing cluster

### PR DESCRIPTION
**Pull request by jlong49**
#675 

Signed-off-by: John Long <john.long@enterprisedb.com>

### CODE REVIEW

**Review by thomasboussekey — APPROVED** [review]

**Review by sxd — APPROVED** [review]

File: api/v1/cluster_types.go [hunk 1]
```diff
@@ -111,6 +111,9 @@ const (
 
 	// PGBouncerPoolerUserName is the name of the role to be used for
 	PGBouncerPoolerUserName = "cnpg_pooler_pgbouncer"
+
+	// NewWalReason is the reason that is set when we do a rolling upgrade to add WAL volumes to a cluster
+	NewWalReason = "the instance has unattached wal volumes"
```
Comment (gbartolini) [inline api/v1/cluster_types.go:116]:
```suggestion
	NewWalReason = "the instance has unattached WAL volumes"
```

**Comment by armru** [comment]
the other commit given that is too large and has a proper meaning has been moved to: https://github.com/cloudnative-pg/cloudnative-pg/pull/1319

File: api/v1/cluster_webhook.go [hunk 2]
```diff
@@ -1199,6 +1199,10 @@ func (r *Cluster) validateWalStorageChange(old *Cluster) field.ErrorList {
 		}
 	}
 
+	if old.Spec.WalStorage == nil && r.Spec.WalStorage != nil {
+		return validateStorageConfigurationSize("walStorage", *r.Spec.WalStorage)
+	}
+
```
Comment (mnencia) [inline api/v1/cluster_webhook.go:1205]: I think the regular validation already performs this check.

**Review by mnencia — APPROVED** [review]

**Pull request merged**